pkg/crdutil/crdutil.go

```diff
@@ -84,38 +92,47 @@ func EnsureCRDsCmd() {
 		log.Fatalf("Failed to create API extensions client: %v", err)
 	}
 
-	if err := walkCrdsDir(ctx, client.ApiextensionsV1().CustomResourceDefinitions()); err != nil {
+	if err := walkCRDs(ctx, client.ApiextensionsV1().CustomResourceDefinitions()); err != nil {
 		log.Fatalf("Failed to apply CRDs: %v", err)
 	}
 }
 
-// walkCrdsDir walks the CRDs directory and applies each YAML file.
-func walkCrdsDir(ctx context.Context, crdClient v1.CustomResourceDefinitionInterface) error {
-	for _, crdDir := range crdsDir {
+// walkCRDs walks the CRDs directory and applies each YAML file.
+func walkCRDs(ctx context.Context, crdClient v1.CustomResourceDefinitionInterface) error {
+	for _, crdDir := range files {
+		// We can skip the errors as it has been checked in initFlags.
+		parentDir, _ := os.Stat(crdDir)
 		// Walk the directory recursively and apply each YAML file.
 		err := filepath.Walk(crdDir, func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
-			if info.IsDir() || filepath.Ext(path) != ".yaml" {
+			if info.IsDir() {
+				return nil
+			}
+			if filepath.Ext(path) != ".yaml" && filepath.Ext(path) != ".yml" {
+				return nil
+			}
+			// If not recursive we want to only apply the CRDs in the top-level directory.
+			if !recursive && parentDir.IsDir() && filepath.Dir(path) != strings.TrimRight(crdDir, "/") {
```

Comment:
should we add a test case for this new branch or should these all be included in the "TODO add unit tests" comment you have added?

Reply:
Let me take another look at the tests, I think I can easily test the `walkCRDs` func